pkg/config/cli_test.go

```diff
@@ -54,4 +54,15 @@ func TestControllerOptions_Normalize(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, expected, underTest.DisableComponents)
 	})
+
+	t.Run("replacesDeprecation", func(t *testing.T) {
+		disabled := []string{"helm", "kubelet-config", "coredns", "kubelet-config", "worker-config", "autopilot"}
```

Comment:
Yeah, this is more "replaceDeprecationAndRemoveDupes". I think it makes sense to have this in one test case, i.e. it tests that replacing the deprecation doesn't leave the new component twice in the resulting slice. I do see the issue with the not absolutely accurate test naming, though. Any advice on a better name, or how to split it up?

Reply:
My concern isn't so much the name and not the test with its current implementation but detecting breaking changes in the future. I think this test case is good and valid but it would make sense also test the string slice without worker-config.

Reply:
Added a third test case as suggested.